### DISCUSSION TITLE
R - Pertimbangan konversi ke ordered factor atau unordered factor dari string "No Data Available".

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#46] [Q&A] [answered]
Misal, ada suatu kolom X  yang terdiri dari  string "Average", "Strong", "Weak", "Very Strong", "Very Weak" dan "NDA". 
Maka saat mau mengubah string tersebut menjadi kategori, bagaimana perlakuan yang diberikan terhadap kategori NDA.
Karena NDA ini adalah No Data Available, apakah bijak jika string pada kolom X diubah menjadi factor terkecil sebagaimana berikut = 

`X <-  as.factor(X)`
`X <-ordered(X, levels=c("NDA","Very Weak", "Weak", "Average", "Strong", "Very Strong"))`

Apakah bisa dianggap sebagai kategori paling kecil seperti di atas? 
Atau sebaiknya dibuat menjadi unordered factor saja dengan metode di bawah? 

`X <- factor(X, ordered=FALSE)`

Namun kalau dibuat unordered, rasanya seperti mengabaikan obvious hierarchy yang ada pada string lainnya dari "Very Weak" ke "Very Strong".

**Comment by VictorNugraha** [comment, marked as answer]
Hi @Hertzspr!

Dalam pembuatan levels yang terdapat pada kategori, sebenarnya tidak ada peraturan baku mengenai bagiamana menyusun levels kategori tersebut. Sehingga dalam penyusunan levels pada sebuah kategori dikembalikan berdasarkan pamahan bisnis. 

Untuk kasus yang dihadapai Bapak, jika terdapat sebuah kategori _NDA (No Data Available)_ memang lebih baik kategori tersebut diberikan levels dibawah kategori _Very Weak_. Hal tersebut dikarenakan jika kategori _NDA_ belum memiliki value apapun yang menjelaskan apakah data tersebut lebih kuat dibandingkan kategori _Very Weak_ ataupun lebih lemah dibandingkan kateogri _Very Strong_.

Selain itu jika pada factor tersebut dihiraukan levelsnya, kami setuju denan pernyataan yang Bapak sudah sampaikan yaitu kita secara tidak langsung mengabaikan hal yang sudah jelas. Maka dari itu kami tidak menganjurkan hal tersebut.

Demikian informasi yang dapat kami sampaikan, semoga menjawab pertanyaan yang ditanyakan. Terima kasih.

**Reply by Hertzspr** [reply]
Terima kasih
